Add possibilty to hide or show email in profile

Edmondi-Kacaj opened the pull request:
Hello everyone,

User email-address visibility should be configurable, So every user must have the possibility to show or hide his/her email _(Except for Admin permission which can see emails for each user even if they are not visible)_. those changes are part of the possibility of showing or hiding an email profile for each user.

- Some of the changes made in this pull request are:  

   1.  Create a new property call "**ccm:showEmail**" in [ccmodel.xml](https://github.com/edu-sharing/Edu-Sharing/commit/76cec1ab65467a2124fb9384e25df7fc6a623973#diff-7d8b6576b0fa6da7ed1f52fe88e4ab0d305c1aa44f1f1c5b0ff192b29e9f5593L556), "**showEmail**" by default must be **false**  or hide
   2.  Create an Object call ["**ProfileSettings**"](https://github.com/edu-sharing/Edu-Sharing/commit/12bf9b10143bed9fd00cdc21ff0e183bc0a4d853#diff-b6f2a107f365347a13947a8e4b497522f6ae12e1bb1b8e3d0ec9502b56d05fabL7)  
   3.  Create  2 endpoint **GET,POST** in [IamApi.java ](https://github.com/edu-sharing/Edu-Sharing/commit/f2a059cbbe5c0a042fcef32f920af1d385a04c6e#diff-a887251e5ec48525e0ad029c990a31d77823e85e51940de65318c0020e53196aL349), to make it possible to communicate with the frontend.
   4. Implement function to retrieve and set values from/to alfresco [PersonDao](https://github.com/Edmondi-Kacaj/Edu-Sharing/blob/add-possibilty-to-hide-or-show-email-in-profile/Backend/src/org/edu_sharing/restservices/PersonDao.java#L557-L578)
   5.  Hide emails for the non-admin user, if the email is set **false** by owner [PersonDao](https://github.com/Edmondi-Kacaj/Edu-Sharing/blob/add-possibilty-to-hide-or-show-email-in-profile/Backend/src/org/edu_sharing/restservices/PersonDao.java#L358), _**Admin** can see email even if they are hidden_
   6. Hide Email for each Node, for every user, who set Email as hidden,  _**Admin** can see email even if they are hidden_  [NodeDao](https://github.com/Edmondi-Kacaj/Edu-Sharing/blob/add-possibilty-to-hide-or-show-email-in-profile/Backend/src/org/edu_sharing/restservices/NodeDao.java#L1274)
   7. Implement logic in Frontend [profiles.component.html Line: 46 and L63-70](https://github.com/Edmondi-Kacaj/Edu-Sharing/blob/add-possibilty-to-hide-or-show-email-in-profile/Frontend/src/app/modules/profiles/profiles.component.html#L46-L70)
   8.  implement Logic in Frontend, [profiles.component.ts L95 and L208  ](https://github.com/Edmondi-Kacaj/Edu-Sharing/blob/add-possibilty-to-hide-or-show-email-in-profile/Frontend/src/app/modules/profiles/profiles.component.ts)


- This pull Request is mus be merged together with merge in  [core-module](https://scm.edu-sharing.com/Angular/core-module/-/merge_requests/2)

Edmondi-Kacaj committed:
Add a new boolean field in alfresco

Edmondi-Kacaj committed:
Add new field into Constants

Edmondi-Kacaj committed:
Implement function to retrive and set values from alfresco db

Edmondi-Kacaj committed:
add Model for this new values , and implemented 2 end-point to get and put value from REST-FULL API

Edmondi-Kacaj committed:
implemented the logic in front-end to check if email must be show or not, and possibility to hide or show email

Edmondi-Kacaj committed:
add Translation for this impementation

Edmondi-Kacaj committed:
refactor property name to showEmail

Edmondi-Kacaj committed:
refactor name and use CammelCase

Edmondi-Kacaj committed:
refactor name and use CammelCase in Frontend part

Edmondi-Kacaj committed:
Minor fix in GET  path

Edmondi-Kacaj committed:
Hide all e-mails for non-admin user,non admin cannot see other emails if they have make unvisible

Edmondi-Kacaj committed:
Refactor code

Edmondi-Kacaj committed:
change core-module branch from develop to release/5_1

Edmondi-Kacaj committed:
change core-ui-module branch from develop to release/5_1

Edmondi-Kacaj committed:
Change subModules from release/5_1 to develop

Edmondi-Kacaj committed:
Use Object ProfileSettings, instead of single property,and Refactor method and property to use this Object

Edmondi-Kacaj committed:
Fix translate for show and hide email

Edmondi-Kacaj committed:
Refactor Model in Backend from ShowEmail to ProfileSettings

Edmondi-Kacaj committed:
Refactor PersonDao and IamApi to use the ProfileSettings Model, instead of ShowEmail Model

Edmondi-Kacaj committed:
Hide Email for each Node, for every user, who set Email as private (Admin will see all email even if they are hidden)

Edmondi-Kacaj committed:
Refactor Typos comments , and methods name

Edmondi-Kacaj committed:
Remove unused method

Edmondi-Kacaj committed:
Refactor profiles component:
  - make switch label static
  - Refactor common/(de,en).json

Edmondi-Kacaj committed:
Hide email in Properties

Edmondi-Kacaj committed:
Reformat Function getProperties

Edmondi-Kacaj committed:
Fix the condition and Format the code and Typos

Edmondi-Kacaj committed:
Merge branch 'develop' of https://github.com/edu-sharing/Edu-Sharing into add-possibilty-to-hide-or-show-email-in-profile

Edmondi-Kacaj committed:
Make showEmail false as default

torsten-simon commented:
Loading a whole `PersonDao` for every node in `NodeDao` seems a bit too expensive, may change it to just reading the appropriate property of the userid without building the whole Dao (Alfrescos performance will suffer too much otherwise)

Instead of `public boolean getShowEmail() {`, please also Wrap it by returning and appropriate configuration object (similar to what `setProfileSettings` already does)

Edmondi-Kacaj committed:
change made in this commit are:
          Adjustments in switch email ,
          refactor in profiles.component.ts, saveProfileSettings method

Edmondi-Kacaj committed:
change made in this commit are:
          change method name from getShowEmail to getProfileSettings and return ProfileSettings object ,
          IamApi refactor 'profileSettings' endpoint GET,PUT to use the method we refactor

Edmondi-Kacaj committed:
Merge branch 'develop' of https://github.com/edu-sharing/Edu-Sharing into add-possibilty-to-hide-or-show-email-in-profile

Edmondi-Kacaj committed:
Add a functionin CCConstant to include all properties for ProfileSettings

Edmondi-Kacaj committed:
Adjustments in PersonDao:
  - remove isCurrentUserAdminOrSameUserAsUserName() method
  - refactor all references  old function getShowEmail() to getProfileSettings()

Edmondi-Kacaj committed:
Adjustments in authority:
  - add new method in AuthorityService.java call  getProfileSettingsProperties to get all properties
  - implement this method in AuthorityServiceImpl.java

Edmondi-Kacaj committed:
Adjustments in NodeDao:
  - Add AuthorityService
  - Remove getShowEmail and isCurrentUserAdminOrSameUserAsUserName and use the AuthorityService for checking if user has right to see the emails

Edmondi-Kacaj committed:
Adjustments in profiles.component.[ts,html]:
  - Add a new property call isCurrentUserAdmin to get the Login permission
  - fix logic, Admin must see all the email in profile even if they are private

Edmondi-Kacaj committed:
Adjustments in NodeDao and PersonDao:
  - clean code in NodeDao
  - format code in PersonDao.java

Edmondi-Kacaj committed:
Adjustments in NodeDao:
  - Refactor line 462, use tab instead of space

Edmondi-Kacaj commented:
Hello @torsten-simon

as we spoke about some adjustments need to implement.

Those adjustments consist in :

-  Add Global CSS style for mat-slide-toggle and I created a merge request in [core-ui-module](https://scm.edu-sharing.com/Angular/core-ui-module/-/merge_requests/2) and must be merged together with this pull request.
-  In profiles component implement  new style and refactor save saveProfileSettings method [see commit](https://github.com/edu-sharing/Edu-Sharing/pull/13/commits/0334ce4aa0c37b7734aec2fe1e5b2e3b3327a0b8)
- Change method name from getShowEmail to getProfileSettings and return ProfileSettings object , and refactor endpoint GET,PUT to use the method we changed [see commit](https://github.com/edu-sharing/Edu-Sharing/pull/13/commits/54b0c9fdcf06f488833384d8bf0e8d2cd7d5d3cc)
- Add a function in CCConstant to include all properties for ProfileSettings [see commit](https://github.com/edu-sharing/Edu-Sharing/pull/13/commits/9979ee159da64cd938c6ee4e871c9c9746a8f069)
- Adjustments in [PersonDao](https://github.com/edu-sharing/Edu-Sharing/pull/13/commits/486e055445408e061732a9527528085e59a5591c)
- Addjustments in [NodeDao](https://github.com/edu-sharing/Edu-Sharing/pull/13/commits/263406320ff9dd3e4ed14d3984fca44a6d56c51b)
- Implement a function in [AuthorityService](https://github.com/edu-sharing/Edu-Sharing/pull/13/commits/96c73f3f7bde0e099663d6d315a7d7f9643bdf85) to get ProfileSettings properties

Edmondi-Kacaj committed:
Merge branch 'develop' of https://github.com/edu-sharing/Edu-Sharing into add-possibilty-to-hide-or-show-email-in-profile

Edmondi-Kacaj committed:
Merger with develop branch, and resolve conflicts
